app/src/main/kotlin/com/wire/android/ui/home/conversations/details/editguestaccess/createPasswordProtectedGuestLink/CreatePasswordGuestLinkViewModel.kt

```diff
@@ -51,33 +50,47 @@ class CreatePasswordGuestLinkViewModel @Inject constructor(
     private val editGuestAccessNavArgs: CreatePasswordGuestLinkNavArgs = savedStateHandle.navArgs<CreatePasswordGuestLinkNavArgs>()
     private val conversationId: QualifiedID = editGuestAccessNavArgs.conversationId
 
-    val passwordTextState: TextFieldState = TextFieldState()
-    val confirmPasswordTextState: TextFieldState = TextFieldState()
     var state by mutableStateOf(CreatePasswordGuestLinkState())
         @VisibleForTesting set
 
-    init {
-        viewModelScope.launch {
-            combine(passwordTextState.textAsFlow(), confirmPasswordTextState.textAsFlow(), ::Pair)
-                .distinctUntilChanged()
-                .collectLatest { (password, confirmPassword) ->
-                    state = state.copy(isPasswordValid = validatePassword(password.toString()).isValid && password == confirmPassword)
-                }
+    suspend fun observePasswordValidation() {
+        combine(
+            state.passwordTextState.textAsFlow(),
+            state.confirmPasswordTextState.textAsFlow(),
+            ::Pair
+        ).distinctUntilChanged().collectLatest {
+            state = state.copy(
+                invalidPassword = false
+            )
```

Comment:
shouldn't we check if `password` and `confirmPassword` are invalid first before `invalidPassword` ?

Reply:
Nope, when the password change we reset if there are any errors and only check if the passwords are valid is when the create button is clicked